### PR TITLE
Deprecate joining over a different number of levels

### LINKED ISSUE #34862: DEPR: Merging on different number of levels
Two dataframes with MultiIndex in columns (shapes are different).

```python
df1_columns = pd.MultiIndex.from_tuples([("A0", "B0", "C0"), ("A1", "B1", "C1")])
df1 = pd.DataFrame([[1, 2], [10, 20]], columns=df1_columns)

df2_columns = pd.MultiIndex.from_tuples([("X0", "Y0"), ("X1", "Y1")])
df2 = pd.DataFrame([[1, 200], [10, 200]], columns=df2_columns)
```

**Case 1**

```python
pd.merge(df1, df2, left_on=[("A0", "B0","C0")], right_on=[("X0", "Y0")])
```

returns

```
   (A0, B0, C0)  (A1, B1, C1)  (X0, Y0)  (X1, Y1)
0             1             2         1       200
1            10            20        10       200
```

It's ok. Columns saved its structure from the both dataframes. They are tuples, but it can be fixed (for example with something like [this](https://gist.github.com/dinya/06e374721993fd387db19ee7f83065a5)).

**Case 2**

```python
pd.merge(df2, df1, left_on=[("X0", "Y0")], right_on=[("A0", "B0","C0")])
```

returns

```
   X0   X1  A0  A1
   Y0   Y1  B0  B1
0   1  200   1   2
1  10  200  10  20
```

When the left DataFrame is a DataFrame with fewer column levels the merge result is questionable: lower levels are dropped.

Is this a bug or a feature (result levels shape must be the same as for the left DataFrame)?

Notes:

1. In both cases the warning is the same ``UserWarning: merging between different levels can give an unintended result ...``.
2. Checked with pandas 0.25.1 and 1.0.4.
3. See [notebook](https://gist.github.com/dinya/7e97a66ba00694fa752e7074baa39861).

---

- [x] I have searched the [[pandas] tag](https://stackoverflow.com/questions/tagged/pandas) on StackOverflow for similar questions.

- [x] I have asked my usage related question on [StackOverflow](https://stackoverflow.com) (see [62490244](https://stackoverflow.com/questions/62490244/merge-drops-lower-levels-in-result-columns-when-the-left-df-has-fewer-multiindex)).

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #34862
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

I think deprecating this is a good idea, I stumbled across this accidentally this week and found this really odd

### CODE REVIEW

File: pandas/core/reshape/merge.py [hunk 1]
```diff
@@ -669,11 +669,11 @@ def __init__(
         # warn user when merging between different levels
         if _left.columns.nlevels != _right.columns.nlevels:
             msg = (
-                "merging between different levels can give an unintended "
-                f"result ({left.columns.nlevels} levels on the left,"
+                "merging between different levels is deprecated and will be removed "
+                f"in a futere version. ({left.columns.nlevels} levels on the left,"
```
Comment (mroeschke) [inline pandas/core/reshape/merge.py:673]: futere -> future

**Review by mroeschke — COMMENTED** [review]
One comment otherwise LGTM

**Comment by jreback** [comment]
lgtm ping when updated

File: pandas/core/reshape/merge.py (hunk 1 above)
Comment (phofl) [reply pandas/core/reshape/merge.py:673]: thx

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**